<?xml version="1.0"?>
<launch> 
 <node pkg="hins_he_driver" type="hins_he_driver" name="hins_he_driver" output="screen" >
  <param name="laser_ip" type="string" value="192.168.1.88"/>    <!--  雷达ip  -->
  <param name="port" type="int" value="8080"/>
  <param name="topic_name" type="string" value="scan"/>
  <param name="frame_name" type="string" value="scan"/>
  <param name="y_direction" type="bool" value="false"/>      <!-- 是否设置雷达的正前方为y轴正方向 false为x轴正方向-->

  <!-- 避障参数 -->>
  <param name="block_enable" type="bool" value="false"/>                 <!--true为开启RO获取避障数据功能  false为关闭RO获取避障数据功能-->
  <param name="service_name" type="string" value="hins_he_area_data"/>   <!--避障数据服务名字-->
  
  <!-- laser param -->
  <!-- 是否改变雷达参数  true为改变  false为不改变-->
  <param name="change_param" type="bool" value="false"/>                     
  <!-- 旋转频率  opt="10, 20"-->             
  <param name="spin_frequency_Hz" type="int" value="10" />       
  <!--  角度分辨率 opt="0.025, 0.050, 0.100, 0.250, 0.500"  -->                         
  <param name="angle_increment" type="string" value="0.050" />      
  <!-- 过滤等级  opt="0～3" -->                  
  <param name="noise_filter_level" type="int" value="1" />                                       
  
  <!--串扰过滤参数-->
  <param name="disturb_filter_enable" type="bool" value="false"/>      <!-- 是否开启串扰过滤功能 -->
  <param name="disturb_filter_threshold" type="int" value="80"/>         <!-- 过滤距离阈值 -->
  <param name="disturb_filter_point_num" type="int" value="5"/>        <!-- 过滤点数 -->

  <!-- shadows filter param -->
  <param name="shadows_filter_level" type="int" value="0" />                                <!-- 防拖尾过滤等级  opt="-1～3"-->
  <param name="shadows_filter_max_angle" value="175"/>       
  <param name="shadows_filter_min_angle" value="5"/>
  <param name="shadows_filter_traverse_step" type="int" value="1"/>
  <param name="shadows_filter_window" type="int" value="5"/>

  <!--保留start_angle～end_angle的数据(角度制-以传感器顶盖上的角度为准)-->
  <param name="start_angle" value="0"/>
  <param name="end_angle" value="360"/>
  
 </node>
<!-- <node pkg="tf" type="static_transform_publisher" name="base_to_laser" args="0.00 0.00 0.00 0 0 0 base_link laser 100" /> -->

  <!-- launch rviz -->
  <!--<node name="laser" pkg="rviz" type="rviz" required="true"
        args="-d $(find hins_he_driver)/launch/laser.rviz" /> -->

</launch>


